File: make_icons.js
const { createCanvas } = require('canvas');
const fs = require('fs');

function drawIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    const s = size / 128;

    // Background
    ctx.fillStyle = '#010101';
    const r = 28 * s;
    ctx.beginPath();
    ctx.moveTo(r, 0); ctx.lineTo(size - r, 0);
    ctx.arcTo(size, 0, size, r, r);
    ctx.lineTo(size, size - r);
    ctx.arcTo(size, size, size - r, size, r);
    ctx.lineTo(r, size);
    ctx.arcTo(0, size, 0, size - r, r);
    ctx.lineTo(0, r);
    ctx.arcTo(0, 0, r, 0, r);
    ctx.closePath();
    ctx.fill();

    ctx.lineWidth = 10 * s;
    ctx.lineCap = 'round';

    // Cyan arc
    ctx.strokeStyle = '#25F4EE';
    ctx.beginPath();
    ctx.arc(60 * s, 58 * s, 24 * s, Math.PI * 0.9, Math.PI * 1.8, false);
    ctx.stroke();
    // Cyan arrow
    ctx.fillStyle = '#25F4EE';
    ctx.beginPath();
    ctx.moveTo(80 * s, 34 * s); ctx.lineTo(94 * s, 46 * s); ctx.lineTo(80 * s, 58 * s);
    ctx.closePath(); ctx.fill();

    // Red arc
    ctx.strokeStyle = '#FE2C55';
    ctx.beginPath();
    ctx.arc(68 * s, 70 * s, 24 * s, Math.PI * 1.9, Math.PI * 0.8, false);
    ctx.stroke();
    // Red arrow
    ctx.fillStyle = '#FE2C55';
    ctx.beginPath();
    ctx.moveTo(48 * s, 94 * s); ctx.lineTo(34 * s, 82 * s); ctx.lineTo(48 * s, 70 * s);
    ctx.closePath(); ctx.fill();

    // White X
    ctx.strokeStyle = '#FFFFFF';
    ctx.lineWidth = 14 * s;
    ctx.beginPath(); ctx.moveTo(36 * s, 36 * s); ctx.lineTo(92 * s, 92 * s); ctx.stroke();
    ctx.beginPath(); ctx.moveTo(92 * s, 36 * s); ctx.lineTo(36 * s, 92 * s); ctx.stroke();

    return canvas.toBuffer('image/png');
}

[16, 48, 128].forEach(size => {
    fs.writeFileSync(`icon${size}.png`, drawIcon(size));
    console.log(`Created icon${size}.png`);
});

File: make_icons.py
from PIL import Image, ImageDraw
import math

def draw_icon(size):
    img = Image.new("RGBA", (size, size), (0, 0, 0, 0))
    d = ImageDraw.Draw(img)
    s = size / 128

    # Black rounded background
    r = int(28 * s)
    d.rounded_rectangle([0, 0, size-1, size-1], radius=r, fill=(1, 1, 1, 255))

    # Draw cyan arc (top arrow going right)
    # Arc from roughly 200deg to 320deg
    sw = max(1, int(10 * s))
    cx, cy = int(60*s), int(58*s)
    rad = int(24*s)
    d.arc([cx-rad, cy-rad, cx+rad, cy+rad], start=160, end=310, fill=(37, 244, 238), width=sw)

    # Cyan arrowhead at end of arc
    px, py_ = int(80*s), int(46*s)
    pts = [(int(80*s), int(34*s)), (int(94*s), int(46*s)), (int(80*s), int(58*s))]
    d.polygon(pts, fill=(37, 244, 238))

    # Draw red arc
    cx2, cy2 = int(68*s), int(70*s)
    d.arc([cx2-rad, cy2-rad, cx2+rad, cy2+rad], start=340, end=130, fill=(254, 44, 85), width=sw)

    # Red arrowhead
    pts2 = [(int(48*s), int(94*s)), (int(34*s), int(82*s)), (int(48*s), int(70*s))]
    d.polygon(pts2, fill=(254, 44, 85))

    # White X
    xw = max(1, int(14 * s))
    d.line([(int(36*s), int(36*s)), (int(92*s), int(92*s))], fill=(255, 255, 255), width=xw)
    d.line([(int(92*s), int(36*s)), (int(36*s), int(92*s))], fill=(255, 255, 255), width=xw)

    return img

for size in [16, 48, 128]:
    img = draw_icon(size)
    img.save(f"icon{size}.png", "PNG")
    print(f"Saved icon{size}.png")
